src/course-unit/data/utils.js

```diff
@@ -50,3 +50,35 @@ export const getNotificationMessage = (type, isVisible, isModalView) => {
 
   return notificationMessage;
 };
+
+/**
+ * Updates the 'id' property of objects in the data structure using the 'blockId' value where present.
+ * @param {Object} data - The original data structure to be updated.
+ * @returns {Object} - The updated data structure with updated 'id' values.
+ */
+export const updateXBlockBlockIdToId = (data) => {
+  if (!data?.children?.length) {
+    return data;
+  }
+
+  const updatedData = { ...data };
+
+  updatedData.children.forEach((item, index) => {
+    if (item.blockId) {
+      updatedData.children[index] = { ...item, id: item.blockId };
+    }
+
+    const { userPartitionInfo } = item;
+    if (userPartitionInfo?.selectablePartitions) {
+      updatedData.children[index].userPartitionInfo = {
+        ...userPartitionInfo,
+        selectablePartitions: userPartitionInfo.selectablePartitions.map(partition => ({
+          ...partition,
+          groups: partition.groups.map(group => (group.id ? { ...group, id: group.blockId } : group)),
+        })),
+      };
+    }
+  });
+
+  return updatedData;
+};
```

Comment:
```suggestion
export const updateXBlockBlockIdToId = (data) => {
  // Base case: if data is not an object or is null, return it unmodified
  if (typeof data !== 'object' || data === null) {
    return data;
  }

  // Process arrays recursively
  if (Array.isArray(data)) {
    return data.map(updateXBlockBlockIdToId);
  }

  // For objects, create a new object to accumulate the results
  const updatedData = {};

  Object.keys(data).forEach(key => {
    const value = data[key];

    // If the current property is 'children' or 'selectablePartitions' or 'groups', process it recursively
    if (key === 'children' || key === 'selectablePartitions' || key === 'groups') {
      updatedData[key] = updateXBlockBlockIdToId(value);
    } else if (key === 'blockId') {
      // Replace 'blockId' with 'id', maintaining other properties
      updatedData['id'] = value;
    } else {
      // Copy other properties unchanged
      updatedData[key] = value;
    }
  });

  // Special handling for objects with both 'id' and 'blockId' to ensure 'blockId' takes precedence
  if ('blockId' in data) {
    updatedData['id'] = data['blockId'];
  }

  return updatedData;
};
```
Check if this is more effecient implementation

Reply:
Added and deleted unnecessary comments
Thanks